zerver/views/user_groups.py

```diff
@@ -107,6 +107,48 @@ def get_user_group(request: HttpRequest, user_profile: UserProfile) -> HttpRespo
     return json_success(request, data={"user_groups": user_groups})
 
 
+def get_parsed_group_setting_value(user_group: UserGroup) -> Union[int, AnonymousSettingGroupDict]:
+    if hasattr(user_group, "named_user_group"):
+        return user_group.id
+
+    return {
+        "direct_members": list(user_group.direct_members.all().values_list("id", flat=True)),
+        "direct_subgroups": list(user_group.direct_subgroups.all().values_list("id", flat=True)),
+    }
+
+
+def is_api_value_same_as_current_setting_value(
+    current_value: Union[int, AnonymousSettingGroupDict],
+    api_value: Union[int, AnonymousSettingGroupDict],
+) -> bool:
+    if isinstance(current_value, int):
```

Comment:
I am not confident of this function here, since it seems somewhat hard to read. Would try to think of something so that we can get a better readable function but we also duplicate a lot of code and queries. Maybe we can add some comments for explaination.

Reply:
I think probably the right approach involves having a bit of code that rewrites the `{direct_members: [], direct_subgroups: [group_id]}` format into the integer format at the API boundary, before we try to look up any objects; possibly even in a `typed_endpoint` parsing function layer, or as basically the first thing we do in the views code -- and then we don't need to consider that type of possibility here.

Reply:
Yeah, seems fine to do that as a follow-up refactor later.